services/dis-apim-operator/test/utils/azure_fake.go

```diff
@@ -0,0 +1,96 @@
+package utils
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/Altinn/altinn-platform/services/dis-apim-operator/internal/azure"
+	azfake "github.com/Azure/azure-sdk-for-go/sdk/azcore/fake"
+	apim "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/apimanagement/armapimanagement/v2"
+	apimfake "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/apimanagement/armapimanagement/v2/fake"
+)
+
+// NewAPIMClient creates a new APIMClient
+func NewFakeAPIMClient(config *azure.ApimClientConfig) (*azure.APIMClient, error) {
+	clientFactory, err := apim.NewClientFactory(config.SubscriptionId, nil, config.FactoryOptions)
+	if err != nil {
+		return nil, err
+	}
+	return azure.NewApimClientWithFactory(config, clientFactory), nil
+}
+
+func GetFakeBackendServer(
+	getResponse apim.BackendClientGetResponse,
+	getErrorCode *int,
+	createOrUpdateResponse apim.BackendClientCreateOrUpdateResponse,
+	createOrUpdateErrorCode *int,
+	deleteResponse apim.BackendClientDeleteResponse,
+	deleteErrorCode *int,
+) *apimfake.BackendServer {
+	fakeServer := &apimfake.BackendServer{
+		CreateOrUpdate: func(
+			ctx context.Context,
+			resourceGroupName string,
+			serviceName string,
+			backendID string,
+			parameters apim.BackendContract,
+			options *apim.BackendClientCreateOrUpdateOptions,
+		) (resp azfake.Responder[apim.BackendClientCreateOrUpdateResponse], errResp azfake.ErrorResponder) {
```

Comment:
Hmm, having the named return values might be confusing here.

Reply:
You are right! I will remove them :+1: